extensions/amp-accordion/1.0/accordion.js

```diff
@@ -253,30 +282,45 @@ export function AccordionSection({
     prefix,
   } = useContext(AccordionContext);
 
+  const expanded = isExpanded ? isExpanded(id, defaultExpanded) : expandedState;
+  const animate = contextAnimate ?? defaultAnimate;
+  const contentId = `${prefix || 'a'}-content-${id}-${suffix}`;
+
   useEffect(() => {
     hasMountedRef.current = true;
     return () => (hasMountedRef.current = false);
   }, []);
 
+  // Storying this state change callback in a ref because this may change
+  // frequently and we do not want to trigger a re-register of the section
+  // each time  the callback is updated
+  const onExpandStateChangeRef = useRef(
+    /** @type {?function(boolean):undefined|undefined} */ (null)
+  );
+  onExpandStateChangeRef.current = onExpandStateChange;
   useLayoutEffect(() => {
     if (registerSection) {
-      return registerSection(id, defaultExpanded);
+      return registerSection(id, defaultExpanded, onExpandStateChangeRef);
```

Comment:
What is the difference between using a ref instead of using the prop directly and excluding it from the dependency array? I know our linter will complain, but aren't we choosing to ignore updates in the value either way?

Reply:
Actually using the ref keeps the `onExpandStateChange` up to date w/o having it in the dependency array.  Since we are passing a reference and not the function itself, the ref will always be up to date even if `onExpandStateChange` changes.

Reply:
Oh good point, awesome :)

Reply:
Of course, thanks for checking!